File: gen_scripts/ruby/f01.loop.rb
#!/usr/bin/env ruby
# testbox: title="Dir.entries with for..in collection"
# collection loop from output of directory listing, without a subshell -
# see f00.loop.rb for the subshell (`ls`) version of this same lesson
entries = Dir.entries("dirtest").reject { |e| e == "." || e == ".." }.sort
for item in entries                       # cycle through directory listing
   if File.directory? "dirtest/#{item}"  # test if path is directory
       puts "#{item} is a directory"
   else
       puts "#{item} is not a directory"
   end
end

File: gen_scripts/python2/f01.loop.py
#!/usr/bin/env python2
# testbox: title="os.listdir() with for..in collection"
import os

# collection loop on directory listing from Python's own os.listdir(),
#  without a subshell - see f00.loop.py for the subshell (`ls`) version
#  of this same lesson
for item in sorted(os.listdir("dirtest")):
  path = "dirtest/" + item
  if os.path.isdir(path):
    print "%s is a directory" % item
  else:
    print "%s is not a directory" % item
